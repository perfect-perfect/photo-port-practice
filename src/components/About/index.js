import coverImage from "../../assets/cover/cover-image.jpg"

function About() {
    return (
        <section className="my-5">
            <h1 id="about">Who am I?</h1>
            <img src={coverImage} className="my-2" style={{ width: "100%" }} alt="cover" />
            <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Aliquam sem et tortor consequat id porta nibh venenatis. Mauris nunc congue nisi vitae suscipit. Pretium aenean pharetra magna ac placerat. Lectus nulla at volutpat diam ut venenatis tellus in metus. Amet est placerat in egestas erat imperdiet sed euismod. Enim ut sem viverra aliquet eget sit. Magna ac placerat vestibulum lectus. Quam vulputate dignissim suspendisse in est ante in nibh. Facilisis volutpat est velit egestas. Nec sagittis aliquam malesuada bibendum arcu vitae elementum curabitur vitae. Ridiculus mus mauris vitae ultricies. Vel risus commodo viverra maecenas accumsan lacus. Imperdiet dui accumsan sit amet nulla facilisi morbi tempus.
            </p>
        </section>
    );
}

export default About;