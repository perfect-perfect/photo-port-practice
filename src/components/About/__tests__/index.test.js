// the imported 'render' function will render the component for testing
// 'cleanup' is used to remove components from the DOM to prevent memory leaking, as well as variable or data collisions between tests
import { render, cleanup } from '@testing-library/react';

// import the extend-expect library from the jest-dom package.
// jest-dom offers access to custom matchers that are used to test DOM elements.
import '@testing-library/jest-dom/extend-expect';

// import the component we will be testing
import About from '..';

// This will ensure that after each test, we won't have any leftover memory data that could give us false results.
afterEach(cleanup);

// we use the 'describe()' function to declare the component we're testing
// renders the About test
describe('About component', () => {
    // First Test
    //  - the first argument is a string that states what is being tested, in this case 'renders'
    //  - the second argument is a callback function that runs the test
    test('renders', () => {
        // we use the render function to render the 'About' component using JSX. 
        render(<About />);
    });

    // Second Test
    // we'll compare snapshot versions of the DOM node structure
    //  a 'snapshot' is a serialized version of the DOM node structure, enabled by Jest.
    test('matches snapshot DOM node structure', () => {
        // render About
        // we'll use the 'asFragment' function, which returns a snapshot of the 'About' component
        const { asFragment } = render(<About />);

        // we'll test and compare whether the expected and actual outcomes match.
        //  - use the 'expect' function with a matcher to assert something about a value
        //  - we'll use the 'toMatchSnapshot' matcher to assert that snapshots will match
        // sample
        // sample
        expect(asFragment()).toMatchSnapshot();
    })
})
