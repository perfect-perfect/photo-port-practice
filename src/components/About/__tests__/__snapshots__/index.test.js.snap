// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <section
    class="my-5"
  >
    <h1
      id="about"
    >
      Who am I?
    </h1>
    <img
      alt="cover"
      class="my-2"
      src="cover-image.jpg"
      style="width: 100%;"
    />
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Aliquam sem et tortor consequat id porta nibh venenatis. Mauris nunc congue nisi vitae suscipit. Pretium aenean pharetra magna ac placerat. Lectus nulla at volutpat diam ut venenatis tellus in metus. Amet est placerat in egestas erat imperdiet sed euismod. Enim ut sem viverra aliquet eget sit. Magna ac placerat vestibulum lectus. Quam vulputate dignissim suspendisse in est ante in nibh. Facilisis volutpat est velit egestas. Nec sagittis aliquam malesuada bibendum arcu vitae elementum curabitur vitae. Ridiculus mus mauris vitae ultricies. Vel risus commodo viverra maecenas accumsan lacus. Imperdiet dui accumsan sit amet nulla facilisi morbi tempus.
    </p>
  </section>
</DocumentFragment>
`;
